desc/vmec.py

```diff
@@ -712,8 +736,29 @@ def save(  # noqa: C901
         jdotb = file.createVariable("jdotb", np.float64, ("radius",))
         jdotb.long_name = "flux surface average of J*B, on full mesh"
         jdotb.units = "N/m^3"
-        jdotb[:] = grid_full.compress(data_full["<J*B>"])
-        jdotb[0] = 0
+        # in VMEC, they use the form of parallel current from
+        # assuming Boozer coordinates, which is what we will also use here.
+        # this can differ a lot from our <J*B> quantity
+        JB = (
+            (
+                data_full["G"] * data_full["I_r"] / data_full["psi_r"]
+                - data_full["G_r"] * data_full["I"] / data_full["psi_r"]
+            )
+            / data_full["sqrt(g)_Boozer"]
+            * data_full["psi_r"]
+        )
+        JB = (
+            surface_averages(
+                grid_full, JB, sqrt_g=data_full["sqrt(g)"], expand_out=False
+            )
+            / mu_0
+        )
+
+        jdotb[:] = JB
+        jdotb[0] = (
+            0  # NOTE: This does not actually match VMEC wouts,
```

Comment:
is this zero physically?

Reply:
![image](https://github.com/user-attachments/assets/e3abd1c2-b767-45ac-85ec-7c3d5b36d8c4)

Since we can compute at 0 and it is not NaN, means that the things it depends on have non-nan axis limits and so seems it exists, I don't really see why it should be zero on axis anyways as J itself is a current density, not a net current. There can exist current flowing along the axis, net current is just zero due to there being vanishing surface area there but this is not a net current.